python/test/test_cpp_algos.py

```diff
@@ -153,6 +154,29 @@ def test_sssp(graph: Graph):
     verify_sssp(graph, start_node, property_name)
 
 
+def test_ksssp(graph: Graph):
+    weight_name = "workFrom"
+    start_node = 0
+    report_node = 10
+    num_paths = 5
+
+    table = ksssp(graph, weight_name, start_node, report_node, num_paths)
+    paths = table.to_dist()["path"]
+    assert len(paths) <= num_paths
+
+    for i, i_path in enumerate(paths):
+        assert paths[i][-1] == report_node
+
+        for _, j_path in enumerate(paths, start=i):
+            if len(i_path) == len(j_path):
+                unique_path = False
+                for (elem_i, elem_j) in zip(i_path, j_path):
+                    if elem_i != elem_j:
+                        unique_path = True
+                        break
+                assert unique_path
```

Comment:
This is complex enough to document what you are doing so people don't have to figure out you are checking that the paths are different.

Also I think you can write this whole thing as follows:
```suggestion
    for i, i_path in enumerate(paths):
        assert paths[i][-1] == report_node
        assert all(list(i_path) == list(j_path) for j_path in paths[i:])
```
`list(x)` pretty much just iterates `x` and puts it's elements in a list. While the paths could in theory be O(|E|), it's unlikely that will be the case for testing.

Reply:
I like the suggestion, I think the main difference would be for the last assert to do `list(i_path) != list(j_path)` given the last check is meant to see if all paths are unique

Reply:
Oh. Yea.That was a dumb on my part. :-)